src/tequila/circuit/_gates_impl.py

```diff
@@ -376,11 +376,44 @@ def extract_targets(generator):
             targets += [k for k in ps.keys()]
         return tuple(set(targets))
 
-    def __init__(self, angle, generator, control=None, eigenvalues_magnitude=0.5, steps=1, assume_real=False):
-        super().__init__(eigenvalues_magnitude=eigenvalues_magnitude, assume_real=assume_real, name="GenRot", parameter=angle, target=self.extract_targets(generator), control=control)
+    def __init__(self, angle, generator, p0=None, control=None, target=None, eigenvalues_magnitude=0.5, steps=1, name="GenRot", assume_real=False):
+        if target == None:
+            target = self.extract_targets(generator)
+        super().__init__(eigenvalues_magnitude=eigenvalues_magnitude, generator=generator, assume_real=assume_real, name=name, parameter=angle, target=target, control=control)
         self.steps = steps
-        self.generator = generator
+        if control is not None:
```

Comment:
p0 can be None (e.g. in the case of SWAP, or standard rotations). Modification only if neccesary/possible

```python
if control is None and p0 is not None:
```

Reply:
in the QubitExcitation implementation before p0 was always nonzero

Reply:
@kottmanj The bug in the `SWAP` gate should be fixed by now. Thanks for noticing!